app/src/main/java/org/simple/clinic/summary/nextappointment/NextAppointmentEffectHandler.kt

```diff
@@ -1,21 +1,34 @@
 package org.simple.clinic.summary.nextappointment
 
 import com.spotify.mobius.rx2.RxMobius
+import dagger.assisted.Assisted
+import dagger.assisted.AssistedFactory
+import dagger.assisted.AssistedInject
 import io.reactivex.ObservableTransformer
 import org.simple.clinic.overdue.AppointmentRepository
 import org.simple.clinic.util.scheduler.SchedulersProvider
-import javax.inject.Inject
 
-class NextAppointmentEffectHandler @Inject constructor(
+class NextAppointmentEffectHandler @AssistedInject constructor(
     private val appointmentRepository: AppointmentRepository,
-    private val schedulersProvider: SchedulersProvider
+    private val schedulersProvider: SchedulersProvider,
+    @Assisted private val uiActions: NextAppointmentUiActions
 ) {
 
+  @AssistedFactory
+  interface Factory {
+    fun create(uiActions: NextAppointmentUiActions): NextAppointmentEffectHandler
+  }
+
   fun build(): ObservableTransformer<NextAppointmentEffect, NextAppointmentEvent> = RxMobius
       .subtypeEffectHandler<NextAppointmentEffect, NextAppointmentEvent>()
       .addTransformer(LoadNextAppointmentPatientProfile::class.java, loadNextAppointmentPatientProfile())
+      .addConsumer(OpenScheduleAppointmentSheet::class.java, ::openScheduleAppointmentSheet, schedulersProvider.ui())
       .build()
 
+  private fun openScheduleAppointmentSheet(effect: OpenScheduleAppointmentSheet) {
+    uiActions.openScheduleAppointmentSheet(effect.patientUuid)
```

Comment:
Why did we create an explicit function for this?

Reply:
For using method reference in the consumer instead of inline lambda